backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/ir/interop/IrProviderForCEnumAndCStructStubs.kt

```diff
@@ -60,15 +66,12 @@ internal class IrProviderForCEnumAndCStructStubs(
     private val cStructClassGenerator =
             CStructVarClassGenerator(context, interopBuiltIns)
 
-    var module: IrModuleFragment? = null
-        set(value) {
-            if (value == null)
-                error("Provide a valid non-null module")
-            if (field != null)
-                error("Module has already been set")
-            field = value
-            value.files += filesMap.values
-        }
+    /**
+     * The final output of this provider is a list of files which contain IR declarations
+     * of enums and structs that should be generated.
+     */
+    val outputFiles: List<IrFile>
+        get() = filesMap.filterKeys { moduleFilter(it.module) }.values.toList()
```

Comment:
Why not filter these on call site?

Reply:
`moduleFilter` will have more uses inside `IrProviderForCEnumAndCStructStubs` later.